### ISSUE TITLE
Gallery preview link on twitter, facebook, etc. (og tags)

### ISSUE THREAD

**Issue by Robinnnnn** [gallery-so/gallery#66]
check out react-snapshot

right now the crawlers can't load dynamic JS (twitter/facebook/google/etc), so the title/preview image/etc. would have to sent off the bat via server. this is the benefit of server-side rendering... and one of the things we lose out on by moving off of nextjs :grimacing: 

there are ways to pre-render the app for crawlers to hit, which we'd have to configure.

in terms of the images themselves, the server would need to create and save a "preview image" whenever someone creates/edits their collection, which @ivan_nyc should be able to do because he's a master of image editing. the image will be stored at some URL, like assets.gallery.com/collectionID, and whenever the user shares their gallery link it'll pull an image from that assets domain (and use a fallback if none exist)

**Comment by kaitoo1** [comment]
Here's a summary of how we could do this:
tldr:
- use Vercel's Open Graph Image as  Service to generate custom images
- Set up dynamic rendering for our app to serve a pre-rendered page with the correct meta tag for crawlers

details:
For relatively low lift image generation, we can use https://github.com/vercel/og-image

We would fork the repo, customize the inputs and template so that the generated image looks the way we want it to, and then deploy it on vercel.
This deploys a service we can use that accepts a bunch of inputs as url params. This allows us to set a url in the page's meta tag with the collection's data for example. The service will generated an image based on the params, or return a cached one if one was generated previously.

In order to set the meta tag on the page in the first place, we need to do dynamic rendering and return a pre-rendered version of the page to crawlers. This is a solved problem so there are several resources we can refer to:     
- https://vascoabelha.com/blog/how-to-use-crawlers-client-side-websites
    - https://developers.google.com/search/docs/guides/dynamic-rendering
